addons/web/static/src/js/components/datepicker.js

```diff
@@ -31,6 +31,9 @@ odoo.define('web.DatePickerOwl', function (require) {
             this.state = useState({ warning: false });
 
             this.datePickerId = `o_datepicker_${datePickerId++}`;
+            this.pickerOptions = Object.assign({
+                format: time.getLangDateFormat(),
+            }, this.props);
```

Comment:
Storing this in a `pickerOptions` attribute would mean that a change to the props would also need to be detected to update the `pickerOptions`, so I would be expecting something in `willUpdateProps`.

What about something like this?
```javascript
constructor(parent, props) {
    const argumentsWithDefaults = Object.assign({
        format: time.getLangDateFormat(),
    }, props);
    super(parent, props);
```

Reply:
Oh right, maybe forget this as it is not a prop any more.

Reply:
You were right; see my comment on your review up above